compoments/report-repair/repair-availability.js

```diff
@@ -7,8 +7,20 @@ import moment from 'moment';
 import {useRouter} from 'next/router';
 
 const RepairAvailability = ({handleChange, values, fromDate}) => {
+  const CreateAppointmentSlotKeyFromDateStrings = (startDateTime, endDateTime) => {
+    return CreateAppointmentSlotKey(moment(startDateTime), moment(endDateTime));
+  }
+
+  const CreateAppointmentSlotKey = (startDateTime, endDateTime) => {
+    return `${startDateTime.unix()}-${endDateTime.unix()}`;
+  }
+  let defaultValue;
+  if (values.availability) {
+    defaultValue = CreateAppointmentSlotKeyFromDateStrings(values.availability?.startDateTime, values.availability?.endDateTime)
```

Comment:
The `value` field stores the current selection in a different form than is stored in the state passed in (`values`), hence the need for this conversion. This fixes an issue in selection upon returning to the availability section after making an initial selection.

Perhaps the correct value should be stored in the state and read back in when this component is revisited(?)
If so, then the summary could remove that value prior to submitting as the API doesn't required it.

Reply:
I've now implemented the above described approach.